src/main/scala/com/raphtory/core/model/graph/visitor/Vertex.scala

```diff
@@ -49,6 +51,38 @@ trait Vertex extends EntityVisitor {
   //individual in edge
   def explodeInEdge(id: Long,after:Long=0L,before:Long=Long.MaxValue): Option[List[ExplodedEdge]]
 
+  // weight
+  private def directedEdgeWeight(dir : Direction = EdgeDirection.Incoming, weightString : String="weight", mergeStrategy : Merge = MergeStrategy.Sum) : Float = {
+    val eWeights = ( dir match {
+      case EdgeDirection.Incoming =>
+        getInEdges()
+      case EdgeDirection.Outgoing =>
+        getOutEdges()
+      case EdgeDirection.Both =>
+        getEdges()
+    })
+      .map(e=>e.weightOrHistory(weightString))
+    mergeStrategy match {
+      case MergeStrategy.Sum => eWeights.sum
+      case MergeStrategy.Max => eWeights.max
+      case MergeStrategy.Min => eWeights.min
+      case MergeStrategy.Product => eWeights.product
+      case _ => 0.0f
+    }
+  }
+  def inWeight(weightString : String="weight", mergeStrategy : Merge = MergeStrategy.Sum) : Float = {
+    directedEdgeWeight(EdgeDirection.Incoming,weightString,mergeStrategy)
+  }
+  def outWeight(weightString : String="weight", mergeStrategy : Merge = MergeStrategy.Sum) : Float = {
+    directedEdgeWeight(EdgeDirection.Outgoing,weightString,mergeStrategy)
+  }
+  def totWeight(weightString : String="weight", mergeStrategy : Merge = MergeStrategy.Sum) : Float = {
+    mergeStrategy match {
+      case MergeStrategy.Difference => directedEdgeWeight(EdgeDirection.Incoming,weightString,MergeStrategy.Sum)
+        - directedEdgeWeight(EdgeDirection.Outgoing,weightString,MergeStrategy.Sum)
+      case _ => directedEdgeWeight(EdgeDirection.Both, weightString, mergeStrategy)
+    }
+  }
```

Comment:
I think it's more intuitive from a graph perspective to use `weight` instead of `strength` cuz depending on your case study, `strength` could mean something completely different.

Reply:
I also agree that `difference`/`balance` should be its own thing rather than packing it here.

Reply:
I thought degree -> unweighted (i.e. count edges), strength -> weighted (i.e. sum edge weights) is pretty standard in graph theory, whereas I would think of node weight as a property of the data (similar to edge weight).

Reply:
We could go with weightedInDegree etc. maybe though personally I would prefer strength

Reply:
Great idea r.e. balance and totally agree about separating out, I did feel a bit cheeky trying to squeeze "difference" in as a merge strategy since it's not really something that is a merge strategy :)